File: plugins/spring-boot/src/main/resources/templates/java/spring-boot/webmvc.g8/src/main/g8/server/src/$if(simple.truthy)$.$else$main$endif$/$if(simple.truthy)$.$else$java$endif$/$package$/server/note/NoteController.java
package $package$.server.note;

import jakarta.validation.Valid;
import java.net.URI;
import java.util.List;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/notes")
public class NoteController {

    private final NoteService notes;

    public NoteController(NoteService notes) {
        this.notes = notes;
    }

    @GetMapping
    public List<NoteDto.View> all() {
        return notes.all();
    }

    @GetMapping("/{id}")
    public NoteDto.View one(@PathVariable long id) {
        return notes.one(id);
    }

    @PostMapping
    public ResponseEntity<NoteDto.View> create(@Valid @RequestBody NoteDto.Create in) {
        NoteDto.View saved = notes.create(in);
        return ResponseEntity.created(URI.create("/api/notes/" + saved.id())).body(saved);
    }
}

File: plugins/spring-boot/src/main/resources/templates/kotlin/spring-boot/webmvc.g8/src/main/g8/server/src/$if(simple.truthy)$.$else$main$endif$/$if(simple.truthy)$.$else$kotlin$endif$/$package$/server/note/NoteController.kt
package $package$.server.note

import jakarta.validation.Valid
import java.net.URI
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/notes")
class NoteController(private val notes: NoteService) {
    @GetMapping
    fun all(): List<NoteView> = notes.all()

    @GetMapping("/{id}")
    fun one(@PathVariable id: Long): NoteView = notes.one(id)

    @PostMapping
    fun create(@Valid @RequestBody input: NewNote): ResponseEntity<NoteView> {
        val saved = notes.create(input)
        return ResponseEntity.created(URI.create("/api/notes/" + saved.id)).body(saved)
    }
}
